integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/DataSourceV2ScanRelationOnStartInputDatasetBuilder.java

```diff
@@ -44,6 +44,12 @@ public boolean isDefinedAtLogicalPlan(LogicalPlan plan) {
 
   @Override
   public List<InputDataset> apply(DataSourceV2ScanRelation plan) {
+    if (context.getSparkExtensionVisitorWrapper() != null
```

Comment:
I'd get rid of this nullcheck

Reply:
right, two tests failed, but I ll adjust them